src/sth_database.js

```diff
@@ -355,7 +416,7 @@
    * @param {Function} callback Callback to inform about any possible error or results
    */
   function getRawData(collection, entityId, entityType, attrName, lastN, hLimit, hOffset,
```

Comment:
Cyclomatic complexity in this function seems huge at first sight. Please, split the function functionality in smaller functions, to improve readibility.